### DISCUSSION TITLE
[SUGGESTION] General Hovercraft improvements

### DISCUSSION THREAD

**Discussion by TIKIRobo** [tsunamayo/Starship-EVO#4030] [Suggestions]
This will be a list of some small QOL improvements to hovercraft.
First. It would be super helpful if hovercraft would exert some sort of downforce. This could be done by if a hover's ride height is set to 4 meters and it goes above that it will being to push down slightly, the higher it goes from that 4m the more force will be exerted, this would not effect hover's in 0g and it would not need additional hover pods (this also could just be that the rings gravity is super weak and needs a buff)

Second. being able to target ships and other hovers. you might already be able to do this, I don't remember, but if you can't it would be nice to be able to

Third. More HUD elements, the hovership's hud is missing key features such as, lead indicator and landing gear status.  

Forth alternate movement mode: when you hit shift (or some other button)  you will enter a different movement mode in which you use the mouse to steer and A and D to strafe, this mode would make aiming a bit harder but would be better for racing hovers i think

**Comment by cloudrambler** [comment]
It would also be great if a ship could be switched between spaceship mode and hovercraft mode. This would make it easier to land fast ships gracefully, by switching to a set of hover thrusters once close to the ground. It would also allow service ships to operate in both modes, such as moving cargo or mining.

**Comment by Uncle-Ulty** [comment]
> It would also be great if a ship could be switched between spaceship mode and hovercraft mode. This would make it easier to land fast ships gracefully, by switching to a set of hover thrusters once close to the ground. It would also allow service ships to operate in both modes, such as moving cargo or mining.

Well, you can do that with a second yoke connected to a hovercraft core.

**Comment by berrigo2** [comment]
Hovercraft should also be able to dock/repair in shipyards the way a spaceship does. Many times in sci-fi, ships bring along accompanying ground vehicles (halo ships + warthogs, mass effect normandy + mako). I would very much appreciate the ability to bring my hovertanks along with me.

Even having another separate block for hovercraft docking would be plenty acceptable.

**Comment by ZachZent** [comment]
One thing that is missing is the ability for hovercraft to dock in shipyards. Probably if it does, it would automatically use landing pad mode.

**Comment by jealousdequan** [comment]
Oooh here's what I want for Christmas: to be able to hold the down button like you do in spaceships, but in hovercrafts, to lower them down to the ground for hovercrafts with super high ride heights.

**Comment by cloudrambler** [comment]
@Uncle-Ulty Agreed, a second yoke would work for the case of dual mode utility ships. But for a main ship coming in to land, by the time you have disengaged from your ship yoke and then engaged with the hovercraft yoke, you could crash. Unless you do the switch at high altitude, which could be dicey for the hover landing anyway. Or if you are in multiplayer and transfer to a co-pilot, that could be cool. But my suggestion was to make the simplest case more elegant, so a hotkey (for example) could switch mode with a minimal delay.

**Comment by Kaiser-Indrasil** [comment]
I'll chip in too, why not.
Being able to **pitch** would help in 3 ways:
- aiming your fixed weapons, which right now are next to useless on hover vehicles, because it's near impossible to hit anything
- bracing for impact when you make a huge jump
- helping at preventing rollovers

Some vehicles are actually so heavy that the Magnet Gun simply can't turn them over. I'd suggest incresing its strength, or, in the long term, temporarily increasing the roll torque of the vehicle itself when it detects it's been rolled over.

When doing test drives around the ringworld habitat, I've noticed that I have to slow down a lot before even small changes in terrain elevation when zooming at relatively medium speeds (70 m/s), otherwise, I will hit the terrain.

Following what others have suggested, there could be an "afterburner" which boosts the repulsor's strength a lot, but the longer it's used, the more energy it uses up, much like shown here in Avorion.

Notice how the energy use increases exponentially and decreases once I stop using the afterburner:

(the video is sped up 200%)

https://user-images.githubusercontent.com/63756057/106187477-999e9a00-619d-11eb-9c4f-4446c6735b95.mp4

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo I've got just 2 more suggestions to easily make hovercraft great again:
- Gimbal assist for fixed weapons. This is the main reason we need pitch control but it can serve as a very nice band-aid.
- Steeper acceleration curve. At present, hovercraft all feel very sluggish when starting from full stop. Their max speed feels alright though, since the habitat presents enough of an off-road challenge as it is.
